lib/spack/spack/spec.py

```diff
@@ -4135,11 +4151,17 @@ def __getitem__(self, name):
                 itertools.chain(
                     # Regular specs
                     (x for x in order() if x.name == name),
+                    (
+                        x
+                        for x in order()
+                        if (not x.virtual)
+                        and any(name in edge.virtuals for edge in x.edges_from_dependents())
+                    ),
```

Comment:
Shouldn't the line after this be unreachable, assuming we've properly constructed virtuals on edges for old specs that we read in?